core/src/main/java/io/grpc/ServerImpl.java

```diff
@@ -130,22 +100,15 @@ synchronized void setTerminationRunnable(Runnable runnable) {
    *
    * @return {@code this} object
    * @throws IllegalStateException if already started
+   * @throws IOException if unable to bind
    */
-  public synchronized ServerImpl start() {
+  public synchronized ServerImpl start() throws IOException {
     if (started) {
       throw new IllegalStateException("Already started");
     }
+    // Start and wait for any port to actually be bound.
+    transportServer.start(new ServerListenerImpl());
```

Comment:
This is the only reference to started, so nothing else will care. I delayed setting started=true in case transportServer throws an exception.